### PR TITLE
Introducing Insights/Perso

### PR DESCRIPTION
**Pull request by Ant-hem**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no     
| Need Doc update   | yes

Introducing the insights feature
 
``` java

SyncUserInsightsClient insights = new SyncInsightsClient("applicationId", "apiKey").user("userToken");

// Click events (you can pass request options to every call
insights.clickedFilters("eventName", "indexName", []string{"filter"});
insights.clickedObjectIDs("eventName", "indexName", []string{"objectID"});
insights.clickedObjectIDsAfterSearch("eventName", "indexName", []string{"objectID"}, []string{42} , "queryID");

// Conversion events
insights.convertedObjectIDs("eventName", "indexName", []string{"objectID"});
insights.convertedObjectIDsAfterSearch("eventName", "indexName", []string{"objectID"}, "queryID");

// View events
insights.viewedFilters("eventName", "indexName", []string{"filter"});
insights.viewedObjectIDs("eventName", "indexName", []string{"objectID"});
```

### CODE REVIEW

**Comment by Ant-hem** [comment]
Working on async insight tomorrow.

File: algoliasearch-common/src/main/java/com/algolia/search/APIClient.java [hunk 1]
```diff
@@ -1750,6 +1751,42 @@ public SearchUserIDs searchUserIDs(
...
+  public PersonalizationStrategyResult getStrategy(@Nonnull RequestOptions requestOptions)
+      throws AlgoliaException {
+
+    return httpClient.requestWithRetry(
+        new AlgoliaRequest<>(
+            HttpMethod.GET,
+            AlgoliaRequestKind.SEARCH_API_READ,
+            Arrays.asList("1", "recommendation", "personalization", "strategy"),
+            requestOptions,
+            PersonalizationStrategyResult.class));
```
Comment (aseure) [inline algoliasearch-common/src/main/java/com/algolia/search/APIClient.java]: I don't know how the strategy features of Personalization will evolve but I'm not sure this name is convenient as a return type of `getStrategy`. Same remark for `setStrategy` response. What about:

* `PersonalizationStrategyResult -> GetStrategyResult` for `getStrategy` response
* `PersonalizationSaveStrategyResult -> SetStrategyResult` for `setStrategy` response

?

File: algoliasearch-common/src/main/java/com/algolia/search/AsyncInsightsClient.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  /**
+   * @param config InsightsClient
+   * @param client APIClient
+   */
+  public AsyncInsightsClient(@Nonnull InsightsConfig config, @Nonnull AsyncAPIClient client) {
+    this.config = config;
+    this.client = client;
+    this.host = "insights." + config.getRegion() + ".algolia.io";
```
Comment (aseure) [inline algoliasearch-common/src/main/java/com/algolia/search/AsyncInsightsClient.java:56]: Is `getRegion` only used to set the Insights region or for Analytics as well? If it's only for Insights, could we rename the parameter and its getter `getInsightsRegion` to ensure we are talking about the Insights API. Also, if it's used for both Analytics and Insights API, I'm wondering if it makes sense to decorrelate them (probably not). What do you think @nunomaduro @julienbourdeau?

File: algoliasearch-common/src/main/java/com/algolia/search/APIClient.java (hunk 1 above)
Comment (Ant-hem) [reply algoliasearch-common/src/main/java/com/algolia/search/APIClient.java]: Yes, make sense. I will rename it. I have the same name in C# should I change it as well?

Comment (Ant-hem) [reply algoliasearch-common/src/main/java/com/algolia/search/APIClient.java]: Fixed with 947d1d1 @aseure

**Pull request merged**